app/src/main/java/de/westnordost/streetcomplete/screens/about/ChangelogFragment.kt

```diff
@@ -52,20 +50,21 @@ class WhatsNewDialog(context: Context, sinceVersion: String) : AlertDialog(conte
 
     init {
         val binding = DialogWhatsNewBinding.inflate(LayoutInflater.from(context))
-        binding.changelogList.addItemDecoration(DividerItemDecoration(context, DividerItemDecoration.VERTICAL))
 
         setTitle(R.string.title_whats_new)
         setView(binding.root)
         setButton(DialogInterface.BUTTON_POSITIVE, context.resources.getText(android.R.string.ok), null, null)
 
         scope.launch {
             val fullChangelog = readChangelog(context.resources)
-            var currentVersionIndex = fullChangelog.indexOfFirst { it.title == sinceVersion }
-            // if version not found, just show the last one
-            if (currentVersionIndex == -1) currentVersionIndex = 1
-            val changelog = fullChangelog.subList(0, currentVersionIndex)
-
-            binding.changelogList.adapter = ChangelogAdapter(changelog)
+            var sinceVersionIndex = fullChangelog.indexOf("<h2>$sinceVersion</h2>")
+            if (sinceVersionIndex == -1) {
+                // if version not found, just show the last one
+                sinceVersionIndex = fullChangelog.indexOf("<h2>", 4)
```

Comment:
This implementation seems a bit unstable too me. If I understand correctly, it intends to finds the second `<h2>` in the document and does this by leaving the first 4 characters out of the search. This breaks if the HTML changelog file does not immediately start with the header of the most current changelog. (Whitespaces?)

A more stable way would be to find the first occurrence, then search again from the first occurrence. Or, maybe more generally, maybe a helper extension function `fun String.indicesOf(): Sequence<Int>` could help to make it better readable?

Reply:
Yeah, you're right. I like the `indicesOf` idea and implemented it in [`1811ddc`](https://github.com/streetcomplete/StreetComplete/pull/5439/commits/1811ddc410f494c205bb24bef812476cef1100a5).